llmfoundry/command_utils/data_prep/convert_text_to_mds.py

```diff
@@ -468,6 +469,11 @@ def convert_text_to_mds(
             trust_remote_code,
         )
 
+    index_path = os.path.join(local_output_folder, 'index.json')
+    with open(index_path, 'r') as index_file:
+        if not json.load(index_file)['shards']:
+            raise ValueError('Input data was too small, no shards written to output folder.')
```

Comment:
@jjanezhang will want this to be a custom error like the other ones i think?

Reply:
Yes! we should add a custom error here telling the user they have too few dataset examples